File: iOS/Mobile/SafariActivity.m
//
//  SafariActivity.m
//  Mobile
//
//  Created by Alan McEwan on 1/7/13.
//  Copyright (c) 2013 Ellucian. All rights reserved.
//

#import "SafariActivity.h"

@implementation SafariActivity
{
	NSURL *_targetURL;
}

- (NSString *)activityType
{
	return @"SafariActivity";
}

- (NSString *)activityTitle
{
	return NSLocalizedString(@"Open in Safari", @"label to open link in Safari");
}

- (UIImage *)activityImage
{
	return [UIImage imageNamed:@"icon_website"];
}

- (BOOL)canPerformWithActivityItems:(NSArray *)activityItems
{
    
	for (id activityItem in activityItems) {
        if ([activityItem isKindOfClass:[NSString class]]) {
            NSURL *url = [NSURL URLWithString:activityItem];
            if([[UIApplication sharedApplication] canOpenURL:url]) {
                return YES;
            }
		}
	}
	
	return NO;
}

- (void)prepareWithActivityItems:(NSArray *)activityItems
{
    //iterate through items and find the url
	for (id activityItem in activityItems) {
        if ([activityItem isKindOfClass:[NSString class]]) {
			_targetURL = [NSURL URLWithString:activityItem];
		}
	}
}

- (void)performActivity
{
	BOOL completed = [[UIApplication sharedApplication] openURL:_targetURL];

	[self activityDidFinish:completed];
}

@end

File: iOS/Mobile/SafariActivity.swift
//
//  SafariActivity.swift
//  Mobile
//
//  Created by Jason Hocker on 8/5/15.
//  Copyright © 2015 Ellucian Company L.P. and its affiliates. All rights reserved.
//

import Foundation

class SafariActivity : UIActivity {
    
    var targetURL : NSURL?
    
    override func activityType() -> String? {
        return "SafariActivity"
    }
    
    override func activityTitle() -> String? {
        return NSLocalizedString("Open in Safari", comment: "label to open link in Safari")
    }
    
    override func activityImage() -> UIImage? {
        return UIImage(named:"icon_website")
    }
    
    override class func activityCategory() -> UIActivityCategory {
        return .Action
    }
    
    override func canPerformWithActivityItems(activityItems: [AnyObject]) -> Bool {
        for item in activityItems {
            if let _ = item as? NSURL {
                return true
            }
            if let urlString = item as? String {
                if let url = NSURL(string: urlString) {
                    if UIApplication.sharedApplication().canOpenURL(url) {
                        return true
                    }
                }
            }
        }
        return false
    }
    
    override func prepareWithActivityItems(activityItems: [AnyObject]) {
        for item in activityItems {
            if let url = item as? NSURL {
                targetURL = url
            } else if let urlString = item as? String, url = NSURL(string:urlString) {
                targetURL = url
            }
        }
    }
    
    override func performActivity() {
        if let targetURL = targetURL {
            let completed = UIApplication.sharedApplication().openURL(targetURL)
            activityDidFinish(completed)
        } else {
            activityDidFinish(false)
        }
    }
}
